### DISCUSSION TITLE
Changing default images in ClusterServiceVersion.yaml

### DISCUSSION THREAD

**Discussion by seema-sandeep-jaisinghani-db** [tektoncd/operator#1542] [Q&A]
Hi Team,

I am trying to use tekton in our project as POC.
We have our own secure image registry we are not allwoed to use registr.redhad.io.
For my sample pipeline, code fails with obvious error as we are not allowed to use public reistries - Failed to pull image "registry.redhat.io/ubi8/ubi-minimal@sha256:6910799b75ad41f00891978575a0d955be2f800c51b955af73926e7ab59a41c3": rpc error: code = Unknown desc = pinging container registry registry.redhat.io: Get "https://registry.redhat.io/v2/":

We have ubi-minimal image in our private registry, is there a way to override this image with registry value somewhere in tekton custom resources?
These Images-
https://github.com/tektoncd/operator/blob/1fc53701027612210cae7a11eb374c381eda104e/operatorhub/openshift/release-artifacts/bundle/manifests/openshift-pipelines-operator-rh.clusterserviceversion.yaml#L852-L905

Much appreciate input on this!

**Comment by jkandasa** [comment]
@seema-sandeep-jaisinghani-db in the tekton-operator deployment, you can override these image as environment variable.
if you are using openshift (and install via subscription) you can override these values on subscription yaml file.

**Comment by seema-sandeep-jaisinghani-db** [comment]
Hi @jkandasa , I updated my task file to set env variables, but still same issue.
Please find task file as below

apiVersion: tekton.dev/v1beta1
kind: Task
metadata:
  name: echo-hello
spec:
  steps:
    - name: echo
      image: docker-registry.default.svc:8000/fabric-images/ubi8-minimal:8.7-1049.1675784874
      env:
        - name: IMAGE_ADDONS_MVN_SETTINGS
          value: docker-registry.default.svc:8000/fabric-images/ubi8-minimal:8.7-1049.1675784874
        - name: IMAGE_PIPELINES_ARG__SHELL_IMAGE
          value: docker-registry.default.svc:8000/fabric-images/ubi8-minimal:8.7-1049.1675784874
      script: |
        #!/usr/bin/env bash
        echo "Hello World"

Logs still says-
Pulling image "registry.redhat.io/ubi8/ubi-minimal@sha256:6910799b75ad41f00891978575a0d955be2f800c51b955af73926e7ab59a41c3"

**Comment by jkandasa** [comment]
@seema-sandeep-jaisinghani-db These images are updated by pipelines-controller and pipelines-controller is managed by tekton-operator. you have to update these images in operator, as I mentioned earlier. 
```
IMAGE_ADDONS_MVN_SETTINGS
IMAGE_PIPELINES_ARG__SHELL_IMAGE
```

**Reply by seema-sandeep-jaisinghani-db** [reply]
@jkandasa  So the operator while installing, we need to update those explicitly in ClusterServiceVersion.yaml

**Comment by jkandasa** [comment]
@seema-sandeep-jaisinghani-db still I am not clear about your installation path.
Are you using bundle(with subscription.yaml) to install the operator? or using `openshift-release.yaml` file? or different approach?

**Reply by seema-sandeep-jaisinghani-db** [reply]
@jkandasa The central team who does installation of operators is different, confirming with them.
